cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -191,16 +145,20 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_offsets(
 
   _group_offsets = std::make_unique<index_vector>(num_keys(stream) + 1, stream);
 
-  auto device_input_table = table_device_view::create(_keys, stream);
-  auto sorted_order       = key_sort_order(stream).data<size_type>();
+  auto preprocessed_keys =
+    cudf::experimental::row::equality::preprocessed_table::create(_keys, stream);
+  auto const comparator =
+    cudf::experimental::row::equality::self_comparator{std::move(preprocessed_keys)};
```

Comment:
Why not create the `comparator` directly from `_keys`? We don't need `preprocessed_keys` more than one time.